### PR TITLE
add where clause from data jpa criteria when reactive

### PR DESCRIPTION
**Pull request by Tcharl**
Reactive did not set 'where' clause in JDBC query when findById was executed, thus throwing an error when when jhipster displayed detail or update page

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/java/package/service/EntityManager.java.ejs [hunk 1]
```diff
@@ -92,11 +93,19 @@ public class EntityManager {
      * @param pageable page parameter, or null, if everything needs to be returned
      * @return sql select statement
      */
-    public String createSelect(SelectFromAndJoin selectFrom, Class<?> entityType, Pageable pageable) {
+    public String createSelect(SelectFromAndJoin selectFrom, Class<?> entityType, Pageable pageable, Criteria criteria) {
         if (pageable != null) {
-            return createSelectImpl(selectFrom.limitOffset(pageable.getPageSize(), pageable.getOffset()), entityType, pageable.getSort());
+            if (null != criteria) {
```
Comment (qmonmert) [inline generators/server/templates/src/main/java/package/service/EntityManager.java.ejs:98]: maybe
``` diff
+ if (criteria != null) {
- if (null != criteria) {
```

**Review by mraible — APPROVED** [review]

**Comment by mraible** [comment]
I tested this branch with the following JDL:

```
application {
  config {
    baseName blog,
    reactive true,
    applicationType monolith,
    authenticationType oauth2,
    packageName com.jhipster.demo.blog,
    prodDatabaseType mysql,
    cacheProvider hazelcast,
    buildTool maven,
    clientFramework angular,
    testFrameworks [protractor]
  }
  entities *
}

entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}

entity Post {
  title String required
  content TextBlob required
  date Instant required
}

entity Tag {
  name String required minlength(2)
}

relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}

relationship ManyToMany {
  Post{tag(name)} to Tag{entry}
}

paginate Post, Tag with infinite-scroll
```

Works great and `npm run e2e` passes!

**Comment by mraible** [comment]
I confirmed it works in a microservices environment as well with the following JDL:

```
application {
  config {
    baseName gateway
    reactive true
    packageName com.okta.developer.gateway
    applicationType gateway
    authenticationType oauth2
    buildTool gradle
    clientFramework vue
    prodDatabaseType postgresql
    serviceDiscoveryType eureka
    testFrameworks [protractor]
  }
  entities Blog, Post, Tag, Product
}

application {
  config {
    baseName blog
    reactive true
    packageName com.okta.developer.blog
    applicationType microservice
    authenticationType oauth2
    buildTool gradle
    prodDatabaseType postgresql 
    serverPort 8081
    serviceDiscoveryType eureka
  }
  entities Blog, Post, Tag
}

application {
  config {
    baseName store
    reactive true
    packageName com.okta.developer.store
    applicationType microservice
    authenticationType oauth2
    buildTool gradle
    databaseType mongodb
    devDatabaseType mongodb
    prodDatabaseType mongodb
    enableHibernateCache false
    serverPort 8082
    serviceDiscoveryType eureka
  }
  entities Product
}

entity Blog {
  name String required minlength(3)
  handle String required minlength(2)
}

entity Post {
  title String required
  content TextBlob required
  date Instant required
}

entity Tag {
  name String required minlength(2)
}

entity Product {
  title String required
  price BigDecimal required min(0)
  image ImageBlob
}

relationship ManyToOne {
  Blog{user(login)} to User
  Post{blog(name)} to Blog
}

relationship ManyToMany {
  Post{tag(name)} to Tag{post}
}

paginate Post, Tag with infinite-scroll
paginate Product with pagination

microservice Product with store
microservice Blog, Post, Tag with blog

deployment {
  deploymentType docker-compose
  appsFolders [gateway, blog, store]
  dockerRepositoryName "mraible"
}
```

**Comment by Tcharl** [comment]
The failing CI steps are not related to reactive at all: should this one be merged?

**Comment by Tcharl** [comment]
FYI, tested with this jdl:

```

application {
  config {
    baseName followup,
    applicationType monolith,
    packageName com.corp,
    authenticationType oauth2,
    devDatabaseType h2Disk,
    prodDatabaseType mssql,
    clientFramework react,
    reactive true,
    testFrameworks [cucumber, cypress],
    enableSwaggerCodegen true,
  }
  entities *
}

entity Member {
  identifier String
  name String
  surname String
  internalMail String pattern(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/)
  platformMail String pattern(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/)
  arrivalDate LocalDate
  departureDate LocalDate
  officialRole Role
  company Company
  hasComputer Boolean
  overrideAverageDailyRate Integer min(0) max(2000)
}

entity Rate {
 company Company,
 role Role,
 location Location,
 averageDailyRate Integer min(0) max(2000)
}

enum Location {
FR,
RO,
MAURITUS
}

enum Role {
 PO("Product owner"),
 PPO("Proxy product owner"),
 SCRUM_MASTER("Scrum master"),
 DEV("Developper"),
 TECH_LEAD("Tech lead"),
 SEC_PARTNER("Security partner"),
 DELIVERY_LEAD("Delivery lead"),
 ENGINEERING_LEAD("Engineering lead"),
 DATA_ENG("Data engineer"),
 PBI_EXPERT("Powerbi designer"),
 UXUI("UX-UI"),
 MODEL_BUILDER("Anaplan model builder")
}

enum Company {
 X,
 Y
}

relationship OneToMany {
  Member{workloads(description)} to Workload{member(name)}
  Product{workloads(description)} to Workload{product(name)}
  Product{cycles(name)} to Cycle{product(name)}
}

entity Product {
  name String
}

entity Cycle {
  name String unique required
  startDate LocalDate
  endDate LocalDate
}

entity Workload {
  description String required unique
  percentage Integer min(1) max(100)
  role Role
}

dto * with mapstruct
service all with serviceImpl

```

**Comment by mraible** [comment]
I'm +1 for merging!

**Pull request merged**